From f567f7868a65d69d8e6ce10b5d882d93df5614ad Mon Sep 17 00:00:00 2001
From: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
Date: Fri, 28 Nov 2025 11:13:17 +0800
Subject: [PATCH] net: ethernet: mtk_eth_soc: change TOP_MISC_NETSYS_PCS_MUX
 offset for NETSYSv3

Since we changed the MT7988's topmisc offset from 0x11D10084 to
0x11D00000 to meet TOPS power control requirements, we also need
to adjust the TOP_MISC_NETSYS_PCS_MUX offset.

Signed-off-by: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.h
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.h
@@ -754,7 +754,7 @@
 #define GEPHY_MAC_SEL          BIT(1)
 
 /* Top misc registers */
-#define TOP_MISC_NETSYS_PCS_MUX	0x0
+#define TOP_MISC_NETSYS_PCS_MUX	0x84
 #define NETSYS_PCS_MUX_MASK	GENMASK(1, 0)
 #define	MUX_G2_USXGMII_SEL	BIT(1)
 #define MUX_HSGMII1_G1_SEL	BIT(0)
